otherlibs/stdlib_alpha/effect.ml

```diff
@@ -623,26 +639,28 @@ module Continuation = struct
      can safely be [Obj.magic]ed from [(c, e, es) res] *)
 
   let get_callstack (Continuation cont) i =
-    get_callstack cont i
+    let bt, cont = get_callstack cont i in
+    bt, Continuation cont
 end
 
 let continue (type a b es)
     (k : (a, b, es) Continuation.t) v (local_ hs) =
   let Continuation (type e c) (cont : (a, c, e, es) continuation) = k in
-  let res : (c, e, es) res = continue cont v hs in
-  (Obj.magic res : b)
+  let res : (c, e, es) res = (continue cont v hs : _ @@ unique) in
+  (* CR mode polymorphism: remove this. *)
+  Obj.magic_unique (Obj.magic res : b)
```

Comment:
I think it would be better to define:
```
val obj_magic_with_unique : 'a @ unique -> 'b @ unique
```
above once, rather than have a lot of `magic_unique` calls.

Reply:
Done.